run_route_scripts/batch.sh

```diff
@@ -18,37 +19,29 @@ function usage() {
 }
 
 #set the input file
-if [ -z "${1}" ]; then
+if [ -z "${1:-}" ]; then
 	usage
-elif [ ! -f "${1}" ]; then
+elif [ ! -f "${1:-}" ]; then
 	usage
 else
-	INPUT="${1}"
+	readonly INPUT="${1}"
 fi
 
 #set config file
-CONF="${2}"
+readonly CONF="${2:-}"
 
 #how many threads do you want, default to max
-if [ "${3}" ]; then
-	CONCURRENCY="${3}"
-else
-    CONCURRENCY=$(nproc)
-fi
+readonly CONCURRENCY="${3:-$(nproc)}"
 
 #where do you want the output, default to current time
-OUTDIR=$(date +%Y%m%d_%H%M%S)_$(basename "${INPUT%.*}")
-if [ "${4}" ]; then
-	OUTDIR="${4}"
-fi
-RESULTS_OUTDIR="results/${OUTDIR}"
-mkdir -p "${RESULTS_OUTDIR}"
-
+readonly DEFAULT_OUTDIR=$(date +%Y%m%d_%H%M%S)_$(basename "${INPUT%.*}")
+readonly RESULTS_OUTDIR="results/${4:-$DEFAULT_OUTDIR}"
+mkdir --parent "${RESULTS_OUTDIR}"
 
 #turn the nice input into something parallel can parse for args
 TMP="$(mktemp)"
-cp -rp "${INPUT}" "${TMP}"
-for arg in $(valhalla_run_route --help | grep -o '\-[a-z\-]\+' | sort | uniq); do
+cp --recursive --preserve=mode,ownership,timestamps "${INPUT}" "${TMP}"
+for arg in $(build/valhalla_run_route --help | grep -o '\-[a-z\-]\+' | sort | uniq); do
```

Comment:
so previously this assumed the binary was on your `PATH` now it requires you to run this from a certain location? im a little confused by this part tbh. isnt this script nested in a dir typically one down from the root. meaning it wouldnt see `build` but rather `../build`?

Reply:
Yea, I was just digging into @dgearhart's comment about `run.sh` and realised both of these scripts rely on the binaries being installed on system path.

Let me restore that behaviour. I tweaked it in order to more easily try different versions without having to install the binaries to the system but I see now that's a larger change than expected.